src/components/implementation/no_interface/scheddev/sl_plugins.h

```diff
@@ -48,4 +48,8 @@ void sl_mod_init(void);
 void sl_timeout_mod_expended(cycles_t now, cycles_t oldtimeout);
 void sl_timeout_mod_init(void);
 
+/* API for thread wakeup using timeout module */
```

Comment:
I just realized that you didn't add a new API for blocking on a timeout.  Am I correct in interpreting this as you not supporting oneshot timeout requests?  I believe this is what cFE will be doing.  I just assumed that there was a `sl_block_timeout(..., timeout)` where timeout is `0` if we want to use the periodic timer (that we will check is already set up), or `X` > 0 for a timeout at an absolute time `X`.

How does this differ from your code?